docs/scripts/link-checker-puppeteer.ts

```diff
@@ -0,0 +1,43 @@
+import { sitePaths } from '../src/data/sitePaths';
+import { checkLink, crawlAllLinks, runArrayPromiseInOrder } from './util';
+
+/**
+ * Divide the sitePaths array so that we can easily run a smaller portion if needed.
+ * e.g. to run the first 10 links. run `yarn docs node --require esbuild-register ./scripts/link-checker-puppeteer.ts 0 10`
+ */
+const start = process.argv[2] ?? 0;
+const end = process.argv[3];
+const testPaths = end ? sitePaths.slice(+start, +end) : sitePaths.slice(+start);
+
+try {
```

Comment:
Can we move this try catch block below where `main` is declared?

Reply:
I renamed `main` but kept it at the end for readability. Open to discuss

Reply:
I personally prefer defining function before it's used, but non-blocking!

Reply:
+1 to moving functions before they are used

Reply:
Done! 4a34c24d0265def8147d3acbc5e25a92d53389ca